static/src/javascripts/projects/common/modules/identity/credentials-api-sign-in.js

```diff
@@ -0,0 +1,54 @@
+// @flow
+import {
+    IdentityCookies,
+    identityFeatures,
+} from 'common/modules/identity/identity-features';
+import { ajaxSignIn } from 'common/modules/identity/api';
+import ophan from 'ophan-tracker-js';
+import { addCookie } from 'lib/cookies';
+
+const ONE_DAY_IN_MILLIS = 86400000;
+
+export const signInWithSavedCredentials = (): Promise<boolean> => {
+    if (identityFeatures.promptForSignIn) {
+        // $FlowFixMe
+        return navigator.credentials
+            .get({
+                password: true,
+                mediation: 'optional',
+            })
+            .then(creds => {
```

Comment:
is JavaScript usually this nested?

Reply:
As far as I can tell yes it is

Reply:
you should be able to split this into smaller functions which would help with the pyramid issue

Reply:
done: 6da4a52